de.dlr.sc.virsat.model.extension.tests.edit.test/src/de/dlr/sc/virsat/model/dvlm/dmf/DmfResourceTest.java

```diff
@@ -659,8 +660,85 @@ protected void doExecute() {
+		
 	}
 	
+	@Test
+	public void testSaveReferencedDObjectDifferentUnloadedContainment() throws IOException {
+		TestStructuralElement beanTestStructuralElement = new TestStructuralElement(concept);
+		TestStructuralElement beanTestStructuralElementOther = new TestStructuralElement(concept);
+
+		ed.getVirSatCommandStack().execute(new RecordingCommand(ed) {
+			@Override
+			protected void doExecute() {
+				resSet.getAndAddStructuralElementInstanceResource(beanTestStructuralElement.getStructuralElementInstance());
+				resSet.getAndAddStructuralElementInstanceResource(beanTestStructuralElementOther.getStructuralElementInstance());
+			}
+		});
+
+		Resource resSei = resSet.getStructuralElementInstanceResource(beanTestStructuralElement.getStructuralElementInstance());
+		Resource resSeiOther = resSet.getStructuralElementInstanceResource(beanTestStructuralElementOther.getStructuralElementInstance());
+		
+		Resource.Factory.Registry resourceRegistry = Resource.Factory.Registry.INSTANCE;
+		Map<String, Object> m = resourceRegistry.getExtensionToFactoryMap();
+		m.put(DmfResource.DMF_FILENAME_EXTENSION, new DmfResourceFactory());
+
+		URI originalSeiUriOther = resSeiOther.getURI();
+		URI dmfSeiUriOther = originalSeiUriOther.appendFileExtension(DmfResource.DMF_FILENAME_EXTENSION);
+		
+		URI originalSeiUri = resSei.getURI();
+		URI dmfSeiUri = originalSeiUri.appendFileExtension(DmfResource.DMF_FILENAME_EXTENSION);
+		
+		ResourceSet dmfResourceSet = new ResourceSetImpl();
+		Resource dmfResourceOther = dmfResourceSet.getResource(dmfSeiUriOther, true);
+		Resource dmfResource = dmfResourceSet.getResource(dmfSeiUri, true);
+		
+		ed.saveAll();
+		
+		DObjectContainer dObjectContainerOther = (DObjectContainer) dmfResourceOther.getContents().get(0);
+		de.dlr.sc.virsat.model.extension.tests.tests.TestCategoryReference dmfCategoryAssignmentReference = TestsFactory.eINSTANCE.createTestCategoryReference();
+		de.dlr.sc.virsat.model.extension.tests.tests.TestCategoryAllProperty dmfCategoryAssignment = TestsFactory.eINSTANCE.createTestCategoryAllProperty();
+
+		dObjectContainerOther.getObjects().add(dmfCategoryAssignment);
+		dmfResourceOther.save(Collections.EMPTY_MAP);
+		
+		DObjectContainer dObjectContainer = (DObjectContainer) dmfResource.getContents().get(0);
+		dmfCategoryAssignmentReference.setTestRefCategory(dmfCategoryAssignment);
+		dObjectContainer.getObjects().add(dmfCategoryAssignmentReference);
+		dmfResource.save(Collections.EMPTY_MAP);
+		
+		//Unload original DVLM resources
+		resSeiOther.unload();
+		resSei.unload();
+		
+		//Check that its still possible to save changes
+		final String changedNameReference = "NewNameReference";
+		final String changedNameReferenced = "NewNameReferenced";
+		dmfCategoryAssignmentReference.setName(changedNameReference);
+		dmfCategoryAssignment.setName(changedNameReferenced);
+		dmfResource.save(Collections.EMPTY_MAP);
+		dmfResourceOther.save(Collections.EMPTY_MAP);
+		

```

Comment:
As far as i understood, by saving, the underlying DVLM should be opened and loaded before the changes are saved to it. Right? Thus, would it make sense, to check if the DVLM resources are wll unloaded before the save statement, and then check if the DVLM resources got correctly laoded?

Reply:
Checking that the resources are properly unloaded makes sense, yes. I add an assert statement for that. Checking if the save operation worked is done by checking if the changes appear in the DVLM resources. I don't think it is necessary to check if the DVLM resource are loaded afterwards as from a DMF perspective it does not matter. Important is that the changes have bean saved.